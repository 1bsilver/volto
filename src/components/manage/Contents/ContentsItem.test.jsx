import React from 'react';
import renderer from 'react-test-renderer';
import { ContentsItemComponent as ContentsItem } from './ContentsItem';

test('renders a contents item component', () => {
  const component = renderer.create(
    <ContentsItem
      item={{
        '@id': '/blog',
        title: 'Blog',
        is_folderish: false,
        '@type': 'Document',
      }}
      selected={false}
      onClick={() => {}}
      indexes={[
        {
          id: 'title',
          type: 'string',
        },
      ]}
      onCut={() => {}}
      onCopy={() => {}}
      onDelete={() => {}}
      onMoveToTop={() => {}}
      onMoveToBottom={() => {}}
      onOrderItem={() => {}}
      connectDragSource={x => x}
      connectDragPreview={x => x}
      connectDropTarget={x => x}
      order={1}
      isDragging={false}
    />,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
