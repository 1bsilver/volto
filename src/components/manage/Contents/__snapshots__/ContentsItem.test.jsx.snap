// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a contents item component 1`] = `
<tr
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <td
    className=""
  >
    <i
      aria-hidden="true"
      className="grey content icon drag handle"
    />
  </td>
  <td
    className=""
  >
    <i
      aria-hidden="true"
      className="black square outline icon"
      onClick={[Function]}
      value="/blog"
    />
  </td>
  <td
    className=""
  >
    <a
      onClick={[Function]}
      style={Object {}}
    >
      <i
        aria-hidden="true"
        className="file text outline icon"
      />
       
      Blog
    </a>
  </td>
  <td
    className=""
  >
    Blog
  </td>
  <td
    className="right aligned"
  >
    <div
      aria-busy={undefined}
      aria-disabled={undefined}
      aria-expanded={false}
      aria-multiselectable={undefined}
      className="ui dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        className="text"
      />
      <i
        aria-hidden="true"
        className="ellipsis vertical icon"
      />
      <div
        className="menu transition left"
      >
        <a
          className="item"
          onClick={[Function]}
          style={Object {}}
        >
          <i
            aria-hidden="true"
            className="write icon"
          />
           Edit
        </a>
        <a
          className="item"
          onClick={[Function]}
          style={Object {}}
        >
          <i
            aria-hidden="true"
            className="eye icon"
          />
           View
        </a>
        <div
          className="divider"
        />
        <div
          aria-checked={undefined}
          aria-disabled={undefined}
          aria-selected={undefined}
          className="item"
          onClick={[Function]}
          role="option"
        >
          <i
            aria-hidden="true"
            className="cut icon"
          />
           Cut
        </div>
        <div
          aria-checked={undefined}
          aria-disabled={undefined}
          aria-selected={undefined}
          className="item"
          onClick={[Function]}
          role="option"
        >
          <i
            aria-hidden="true"
            className="copy icon"
          />
           Copy
        </div>
        <div
          aria-checked={undefined}
          aria-disabled={undefined}
          aria-selected={undefined}
          className="item"
          onClick={[Function]}
          role="option"
        >
          <i
            aria-hidden="true"
            className="trash icon"
          />
           Delete
        </div>
        <div
          className="divider"
        />
        <div
          aria-checked={undefined}
          aria-disabled={undefined}
          aria-selected={undefined}
          className="item"
          onClick={[Function]}
          role="option"
        >
          <i
            aria-hidden="true"
            className="long arrow up icon"
          />
           
          Move to top of folder
        </div>
        <div
          aria-checked={undefined}
          aria-disabled={undefined}
          aria-selected={undefined}
          className="item"
          onClick={[Function]}
          role="option"
        >
          <i
            aria-hidden="true"
            className="long arrow down icon"
          />
           
          Move to bottom of folder
        </div>
      </div>
    </div>
  </td>
</tr>
`;
